node-src/lib/getConfiguration.ts

```diff
@@ -0,0 +1,64 @@
+import { readFile } from 'jsonfile';
+import { ZodError, z } from 'zod';
+import { missingConfigurationFile } from '../ui/messages/errors/missingConfigurationFile';
+import { unparseableConfigurationFile } from '../ui/messages/errors/unparseableConfigurationFile';
+import { invalidConfigurationFile } from '../ui/messages/errors/invalidConfigurationFile';
+
+const configurationSchema = z
+  .object({
+    projectId: z.string(),
+    projectToken: z.string(),
+
+    onlyChanged: z.union([z.string(), z.boolean()]),
+    onlyStoryFiles: z.array(z.string()),
+    onlyStoryNames: z.array(z.string()),
+    untraced: z.array(z.string()),
+    externals: z.array(z.string()),
+    ci: z.boolean(),
+    debug: z.boolean(),
+    junitReport: z.union([z.string(), z.boolean()]),
+    autoAcceptChanges: z.union([z.string(), z.boolean()]),
+    exitZeroOnChanged: z.union([z.string(), z.boolean()]),
+    exitOnceUploaded: z.union([z.string(), z.boolean()]),
+    ignoreLastBuildOneBranch: z.string(),
+
+    buildScriptName: z.string(),
+    outputDir: z.string(),
+
+    storybookBuildDir: z.string(),
+    storybookBaseDir: z.string(),
+    storybookConfigDir: z.string(),
+
+    ownerName: z.string(),
```

Comment:
@tmeasday can you provide some context on the `ownerName` and `repositorySlug` options as these seem more for us for internal use and if that's the case we shouldn't have them documented.

Reply:
Hmm, so it looks like the CLI has `--repository-slug` and the action has these two:

https://github.com/chromaui/chromatic-cli/blob/544464cdf53e8bfe73b7e1781e3834c4eeee3e48/CHANGELOG.md#L119

I'm not sure what makes more sense to be configured in a JSON file. WDYT? @ghengeveld?

Reply:
Also -- I assume options provided to the action should take precedence over configuration settings (that's what I did with CLI flags). Do we agree?

Reply:
Actually @ghengeveld I realise that I am not actually preferencing CLI flags over configuration options yet in this PR. It's a bit complicated, can we pair on it please?

Reply:
Yes, let's pair on it tomorrow?

Reply:
I would allow setting `repositorySlug` and `branchName` rather than `ownerName`. The ownerName can be inferred from either of them (repositorySlug `chromaui/chromatic` and branchName `chromaui:main` both yield ownerName `chromaui`). Usually both are retrieved from Git. I suppose there's value in being able to only override the `ownerName` part, but that would add complexity as we'd have to update the `slug` field on the announce mutation to take the override into account.

Reply:
We decided it's best to drop both options for now because if someone forks the repo they'll keep the configuration, which might be bad.